Automattic/wp-calypso
# Jetpack Onboarding: Handle, set and persist currently onboarded site

This PR introduces some UI state to handle and contain the site that is currently in the Jetpack Onboarding flow. This requires a separate treatment, because the site is not a real full featured site object from `state.sites.items`, rather we only have the site ID and a couple of more things about it.

It updates the onboarding flow to redirect to a page with a site slug initially, using the new UI state to actually match the site slug from the route to an ID, and set that ID as the currently onboarded site ID. Furthermore, it uses that site ID in the `getJetpackOnboardingSiteSlug` selector, which is used in the `JetpackOnboardingMain` to persist the current site slug in the JPO flow.

Contains an alternative to #20860 that works with JPO credentials we actually have vs. using `state.sites.items`.

To test:
* Checkout this branch on your local Calypso.
* Checkout the latest master on your Jetpack sandbox.
* Apply D7342-code on your wp.com sandbox and make sure you're sandboxing the API and jetpack.wordpress.com.
* Go to https://YourJetpackSandbox.com/wp-admin/admin.php?page=jetpack&action=onboard&calypso_env=development where `YourJetpackSandbox.com` is the domain of your Jetpack sandbox.
* Verify that after a couple of redirects you land in http://calypso.localhost:3000/jetpack/onboarding/yourjetpacksandbox.com
* Skip some steps and verify the site slug stays in the route.
